src/providers/ogr/qgsogrconnpool.h

```diff
@@ -61,7 +61,13 @@ class QgsOgrConnPoolGroup : public QObject, public QgsConnectionPoolGroup<QgsOgr
     Q_OBJECT
 
   public:
-    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ) { initTimer( this ); }
+    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ), mRefCount( 0 ) { initTimer( this ); }
+    void ref() { ++mRefCount; }
```

Comment:
No:
- The provider always takes one reference
- Each feature source takes one reference
=> So number of taken references is nProviders(=1) + nActiveFeatureSources